kernel/src/syscall.rs

```diff
@@ -4,10 +4,67 @@
 
 //! Mechanisms for handling and defining system calls.
 //!
-//! This includes:
-//! - syscall class
-//! - error types
-//! - interface trait for context switches
+//! # System Call Overview
```

Comment:
Is this duplicating documentation we have elsewhere and we would be possibly better off just linking to?

(I recognize that this is just moving an existing comment, but I'm wondering if something so lengthy, and more general in description is better suited to the book or similar)

Reply:
I think it is duplicated, but it helps provide context for the file.

Reply:
I agree with @bradjc, I prefer this to be in inline docs _as well_